feat(npm)!: support yarn catalogs (#38215)
renovatebot/renovate: lib/modules/manager/npm/readme.md

- `resolutions`
- `pnpm.overrides`
- `pnpm.catalog.<name>`, such as `pnpm.catalog.default` and `pnpm.catalog.myCatalog`. [Matches any default and named pnpm catalogs](https://pnpm.io/catalogs#defining-catalogs).

> - `yarn.catalog.<name>`, such as `yarn.catalog.default` and `yarn.catalog.myCatalog`. [Matches any default and named yarn catalogs](https://yarnpkg.com/features/catalogs).

Reviewer: ```suggestion
- `yarn.catalog` or `yarn.catalogs.<name>`. [Matches any default and named yarn catalogs](https://yarnpkg.com/features/catalogs).
```
